tests/ui/impl-trait/in-trait/method-signature-matches.lt.stderr

```diff
@@ -11,12 +11,12 @@ note: type in trait
    |
 LL |     fn early<'early, T>(x: &'early T) -> impl Sized;
    |                            ^^^^^^^^^
-   = note: expected signature `fn(&T)`
-              found signature `fn(&'late ())`
+   = note: expected signature `fn(&'early T)`
+              found signature `fn(&())`
 help: change the parameter type to match the trait
    |
-LL |     fn early<'late, T>(_: &T) {}
-   |                           ~~
+LL |     fn early<'late, T>(_: &'early T) {}
```

Comment:
You may think, wow this diagnostic sucks; it's mentioning a lifetime that isn't even in the signature. 

Well, it does suck, but it doesn't suck any more than the error you would get from the code if it had not RPITITs. So I don't think it needs fixing here.